apps/api/src/app/vercel-integration/usecases/setup-integration/setup-integration.usecase.ts

```diff
@@ -0,0 +1,149 @@
+import { HttpService } from '@nestjs/axios';
+import { Injectable, Inject } from '@nestjs/common';
+import { lastValueFrom } from 'rxjs';
+import { GetApiKeysCommand } from '../../../environments/usecases/get-api-keys/get-api-keys.command';
+import { GetApiKeys } from '../../../environments/usecases/get-api-keys/get-api-keys.usecase';
+import { GetEnvironment, GetEnvironmentCommand } from '../../../environments/usecases/get-environment';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+import { SetupIntegrationCommand } from './setup-integration.command';
+
+@Injectable()
+export class SetupIntegration {
+  @Inject()
+  private getApiKeyUsecase: GetApiKeys;
+  @Inject()
+  private getEnvironmentUsecase: GetEnvironment;
+
+  constructor(private httpService: HttpService) {}
+
+  async execute(command: SetupIntegrationCommand): Promise<{
+    success: boolean;
+  }> {
+    try {
+      const tokenData = await this.getVercelToken(command.code);
+
+      const projects = await this.getVercelProjects(tokenData.accessToken, tokenData.teamId);
+
+      const projectIds = projects.map((project) => project.id);
+
+      const envKeys = await this.getEnvKeys(command);
+
+      await this.setEnvironments({
+        projectIds,
+        identifier: envKeys.clientKey,
+        secretKey: envKeys.privateKey,
+        teamId: tokenData.teamId,
+        token: tokenData.accessToken,
+      });
+
+      return {
+        success: true,
+      };
+    } catch (error) {
+      throw new ApiException(
+        error?.response?.data?.error_description || error?.response?.data?.message || error.message
+      );
+    }
+  }
+
+  async getVercelToken(code: string) {
+    const postData = new URLSearchParams({
+      code,
+      client_id: process.env.VERCEL_CLIENT_ID || 'oac_wwSTKHT6nQ2hdosgHavoOG2U',
+      client_secret: process.env.VERCEL_CLIENT_SECRET || 'NLuf2j7avAVJLh8VhLzxs4M1',
+      redirect_uri: process.env.VERCEL_REDIRECT_URI || 'http://localhost:4200/auth/login',
+    });
```

Comment:
```suggestion
    });

```

Blank line here for readability